app/src/components/voting/CastBallot.js

```diff
@@ -0,0 +1,147 @@
+import {encryptVote} from './VoteEncrypt';
+
+/*
+Functional component
+*/
+
+function CastBallot(){
+
+    const [loading,electionRetrieved,,electionData] =  useRetrieveElection(localStorage.getItem('electionIDs'), sessionStorage.getItem('token')); 
+    const [context, ] = useContext(LanguageContext);
+
+    const castBallotEndPoint = "/evoting/cast";
+    const edCurve = kyber.curve.newCurve("edwards25519")
+
+    const [choice, setChoice] = useState('');
+    const [errors, setErrors] = useState({});
+    const [lastVote, setLastVote] = useState('');
+    const [showModal, setShowModal] = useState(false);
+
+
+    useEffect(()=> {
+        fetchItems();
+    }, []); 
+
+
+    const fetchItems = async() => {  
+        let choiceCached = sessionStorage.getItem('myVote');
+        setChoice(choiceCached);
+        setLastVote(choiceCached);
+
+    } 
+
+    const handleCheck = (choiceSelected) =>{
+            setChoice(choiceSelected);
+    }
+
+    /*Transform a string of type "1,2,3" to an array [1,2,3]*/ 
+    function unpack(str) {
+        var bytes = [];
+        var b  =str.split(",");
+        
+        for(var i = 0; i < b.length; i++) {
+            var char = parseInt(b[i]);
+            bytes.push(char);
+        }
+
+        return new Uint8Array(bytes);
+    }
+
+   
+    
+    const sendBallot = async() =>{
+        const ballot = {};
+       
+        sessionStorage.setItem('myVote', choice);
+
+        const [K,C] = encryptVote(choice,Buffer.from(unpack(sessionStorage.getItem('pubKey')).buffer), edCurve);
+
+        const KBuff = K.toProto();
+        const CBuff = C.toProto();
+        //transform buffer to []number
+        const vote = [...Buffer.concat([KBuff,CBuff])].map(x => parseInt(x,10));
+
+        ballot['ElectionID'] = localStorage.getItem('electionIDs'); //TODO: don't store id but api call
+        ballot['UserId'] = sessionStorage.getItem('id');       
+        ballot['Ballot'] = vote;
+        ballot['Token'] = sessionStorage.getItem('token');
+        console.log(ballot);
+        //sending the ballot to evoting server
+        try{
+            const response = await fetch(castBallotEndPoint, {
+                method: 'POST',
+                body: JSON.stringify(ballot)
+            });
+        
+            if(response.ok){
+            const data = await response.json();
+            setShowModal(prev => !prev); 
+            console.log(data);
+            return ;
+            } else{
+                //TODO : maybe show a model for an error
+                console.log("ERROR WHEN CASTING BALLOT!")
+                return (-1);
+            }
+
+ 
+        } catch(e) {
+            
+            return e;
+        }
+
+
+    }
+
+    const handleClick = () => {
+        let errors = {};
+        if(choice === ''){
+            errors['noCandidate'] = Translations[context].noCandidate
+            setErrors(errors);
+            return;
+        }
+        setLastVote(choice);
+        sendBallot();
```

Comment:
what if that function throws an error ? Something should be displayed to the user then.